### PR TITLE
Added Azure API Center resource manager.

### PR DESCRIPTION
**Pull request by azaslonov**
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
            **Moving it from RPSaaSMaster branch of private repo. See corresponding signed-off pull request**:
            https://github.com/Azure/azure-rest-api-specs-pr/pull/12365
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by visingla-ms** [comment]
SignedOff based on Private Repo PR ARM SignOff: https://github.com/Azure/azure-rest-api-specs-pr/pull/12365.

**Comment by JackTn** [comment]
@azaslonov do you need to merge this pr now ?

**Comment by azaslonov** [comment]
> ### Generated ApiView
> Language	Package Name	ApiView Link
> Swagger	Microsoft.ApiCenter	https://apiview.dev/Assemblies/Review/58a1f32cb09a489aab429ef3e62483c7
> Go	sdk/resourcemanager/apicenter/armapicenter	Create ApiView failed. Please ask PR assignee for help
> Java	azure-resourcemanager-apicenter	https://apiview.dev/Assemblies/Review/4bf5b37a62cb474889361a4c0fb1e76a



> @azaslonov do you need to merge this pr now ?

If possible, yes please.

I see this check fails by some reason despite we have Canary deployment:
image

**Comment by azaslonov** [comment]
> ### Generated ApiView
> Language	Package Name	ApiView Link
> Swagger	Microsoft.ApiCenter	https://apiview.dev/Assemblies/Review/58a1f32cb09a489aab429ef3e62483c7
> Go	sdk/resourcemanager/apicenter/armapicenter	Create ApiView failed. Please ask PR assignee for help
> Java	azure-resourcemanager-apicenter	https://apiview.dev/Assemblies/Review/4bf5b37a62cb474889361a4c0fb1e76a
> JavaScript	@azure/arm-apicenter	https://apiview.dev/Assemblies/Review/b83bd43fc73649a996b94f3c3dfb8a24

@Wzb123456789, could you please help with this:
image

**Comment by azaslonov** [comment]
> Hi, @azaslonov, our workflow has detected that there is no management SDK ever released for your RP, to further process SDK onboard for your RP, you should have the SDK client library name of your RP reviewed and approved. Action Required:
> 
> * Follow this guidance [Naming for new initial management or client libraries (new SDKs) - Overview (azure.com)](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/821/Naming-for-new-initial-management-or-client-libraries-(new-SDKs)) to create an issue for management client library name arch board review.
> * Paste the issue link that you created in step 1 in this PR
> 
> Impact: SDK release owner will take the approved management client library name to release SDK. No client library name approval will leads to SDK release delayed.

Here's the link: https://github.com/Azure/azure-sdk-pr/issues/1183

**Comment by JackTn** [comment]
> > ### Generated ApiView
> > Language	Package Name	ApiView Link
> > Swagger	Microsoft.ApiCenter	https://apiview.dev/Assemblies/Review/58a1f32cb09a489aab429ef3e62483c7
> > Go	sdk/resourcemanager/apicenter/armapicenter	Create ApiView failed. Please ask PR assignee for help
> > Java	azure-resourcemanager-apicenter	https://apiview.dev/Assemblies/Review/4bf5b37a62cb474889361a4c0fb1e76a
> > JavaScript	@azure/arm-apicenter	https://apiview.dev/Assemblies/Review/b83bd43fc73649a996b94f3c3dfb8a24
> 
> @Wzb123456789, could you please help with this: image

@azaslonov here is the result https://github.com/Azure/azure-rest-api-specs/pull/24130#issuecomment-1560233898

**Comment by azaslonov** [comment]
> @azaslonov here is the result [#24130 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/24130#issuecomment-1560233898)

@JackTn are we good to merge?

**Comment by JackTn** [comment]
> > @azaslonov here is the result [#24130 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/24130#issuecomment-1560233898)
> 
> @JackTn are we good to merge?

@msyyc take a look on this pr and help to merge it, Thanks

**Review by msyyc — APPROVED** [review]

File: specification/apicenter/resource-manager/Microsoft.ApiCenter/preview/2023-07-01-preview/apicenter.json [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        "properties": {
+          "$ref": "#/definitions/ServiceProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "description": "The service properties to be updated."
+    }
+  },
+  "parameters": {
+    "ServiceName": {
```
Comment (MaryGao) [inline specification/apicenter/resource-manager/Microsoft.ApiCenter/preview/2023-07-01-preview/apicenter.json:431]: @azaslonov The `serviceName` parameter defined here would result in a client-level parameter in SDK side, could you confirm that this is a client-level parameter or this is a method-level parameter shared by most operations?

If this is the latter one, please add the [x-ms-parameter-location](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-parameter-location) as `method`.

```json
"x-ms-parameter-location": "method" <<<<<<<<<<<<<<<<<<<<<<<<<<<
```

Comment (azaslonov) [reply specification/apicenter/resource-manager/Microsoft.ApiCenter/preview/2023-07-01-preview/apicenter.json:431]: @MaryGao, you're right. Fixed in this PR: https://github.com/Azure/azure-rest-api-specs/pull/25344

**Pull request merged**